kong/tools/public.lua

```diff
@@ -53,6 +53,11 @@ do
 
     [MIME_TYPES.json] = function()
       local raw_body  = ngx_req_get_body_data()
+      if not raw_body then
+        ngx_log(ERR, "could not read request body")
```

Comment:
I have been thinking about this a bit more, and I think that for clarity we should say:
```
"could not read request body (ngx.req.get_body_data() returned: nil)"
```

That will help developers and users grasping and debugging what is going on by giving them a pointer at first sight.

Reply:
great